packages/@eventual/aws-runtime/src/handlers/bucket-handler-worker.ts

```diff
@@ -0,0 +1,65 @@
+import serviceSpec from "@eventual/injected/spec";
+// the user's entry point will register streams as a side effect.
+import "@eventual/injected/entry";
+
+import {
+  createBucketNotificationHandlerWorker,
+  getLazy,
+  promiseAllSettledPartitioned,
+} from "@eventual/core-runtime";
+import { S3Handler } from "aws-lambda";
+import {
+  createBucketStore,
+  createEntityClient,
+  createServiceClient,
+} from "../create.js";
+import {
+  bucketName,
+  bucketHandlerName,
+  serviceName,
+  serviceUrl,
+} from "../env.js";
+
+const worker = createBucketNotificationHandlerWorker({
+  bucketStore: createBucketStore(),
+  entityClient: createEntityClient(),
+  serviceClient: createServiceClient({}),
+  serviceName,
+  serviceSpec,
+  serviceUrl,
+});
+
+export default (async (event) => {
+  const records = event.Records;
+  console.log("records", JSON.stringify(records, undefined, 4));
+
+  const results = await promiseAllSettledPartitioned(
+    records,
+    async (record) => {
+      await worker({
+        bucketName: getLazy(bucketName),
+        handlerName: getLazy(bucketHandlerName),
```

Comment:
i forget, what does `getLazy` mean? the name doesn't make sense to me .. this is not a promise, what's lazy about it

Reply:
Same thing it does in cdk. A getter for a value. Added when we moved the env variable assertions to be lazy instead of water.

Reply:
Lazy just means a value that isn't immediately resolved, as decided by the caller, it doesn't imply a promise, future, or a thread as controlled by the sub-process itself.